Sources/Hyperspace/Request/NetworkRequest.swift

```diff
@@ -113,6 +113,18 @@ public extension NetworkRequest {
         
         return request
     }
+    
+    func modifyingHeaders(_ headers: [HTTP.HeaderKey: HTTP.HeaderValue]?) -> Self {
```

Comment:
Three things here:
1. what about adding additional headers instead of replacing all of them
2. modifyingHeaders seems like it could be renamed to either modifyHeaders or replaceHeaders
3. I assume you can pass in nil so it clears your headers and this is the same with body, so how about some method comments

Reply:
1) can add this.
2) this is following the API guidelines. modifyHeaders would be for operating on an inout NetworkRequest, the 'ing' gerund implies a copy will be returned.
3) You can do that, I will add comments and a changelog entry

Reply:
awesome

Reply:
I kind of agree that the name `modifyingHeaders` is somewhat ambiguous. What about `usingHeaders` since the way it's setup (for this method at least) is to replace any existing headers with whatever is included in the dictionary passed in? Maybe `replacingHeaders(with headers: ...)` could be a good name too? I feel like it's hard to tell until you see it at the call site. Some experimentation below:
```
let newRequest = request.modifyingHeaders([.contentType: .applicationJSON])
```

```
let newRequest = request.usingHeaders([.contentType: .applicationJSON])
```

```
let newRequest = request.replacingHeaders(with: [.contentType: .applicationJSON])
```